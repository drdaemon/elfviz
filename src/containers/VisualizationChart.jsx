import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Measure from 'react-measure';

import SunburstChart from '../components/SunburstChart';
import IcicleChart from '../components/IcicleChart';

import * as config from '../constants/config';

class VisualizationChart extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dimensions: {
        width: 0,
        height: 0,
      },
    };
  }

  onResize(contentRect) {
    this.setState({ dimensions: contentRect.bounds });
  }

  render() {
    const { type, data } = this.props;
    const { width } = this.state.dimensions;
    const { height } = config;

    if (data === null) return <div className="text-center alert">No data loaded.</div>;

    return (
      <Measure bounds onResize={this.onResize.bind(this)}>
        {({ measureRef }) =>
          <div ref={measureRef}>
            <h2>Visualization <small>({type})</small></h2>
            {type === 'Icicles' ? <p>Click anywhere to zoom in, or click on the top bar to zoom out.</p> : null}
            {type === 'Icicles' ? <IcicleChart id="chart-icicle" width={width} height={height} data={data} /> : null}
            {type === 'Sunburst' ? <SunburstChart id="chart-sunburst" width={width} height={height} data={data} /> : null}
          </div>
        }
      </Measure>
    );
  }
};

VisualizationChart.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.array,
}

export default VisualizationChart;
