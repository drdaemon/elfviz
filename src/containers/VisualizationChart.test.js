import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import VisualizationChart from './VisualizationChart';

describe('VisualizationChart', () => {
  it('renders without crashing', () => {
    shallow(<VisualizationChart type={''} data={null} />);
  });

  it('should match an empty snapshot', () => {
    const wrapper = shallow(<VisualizationChart type={'Icicle'} data={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match a generic snapshot', () => {
    const wrapper = shallow(<VisualizationChart type={'Icicle'} data={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
