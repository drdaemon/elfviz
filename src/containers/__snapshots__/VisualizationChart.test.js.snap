// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualizationChart should match a generic snapshot 1`] = `
<_class
  bounds={true}
  onResize={[Function]}
/>
`;

exports[`VisualizationChart should match an empty snapshot 1`] = `
<div
  className="text-center alert"
>
  No data loaded.
</div>
`;
